add kube-builder operators dashboard

hervenicol opened the pull request:
Towards https://github.com/giantswarm/giantswarm/issues/29262

This PR adds a dashboard that helps diagnosing kube-builder-based operators.

Screenshots:
![image](https://github.com/giantswarm/dashboards/assets/12008875/a402d264-3e45-4287-be65-0975ff3b5d62)
![image](https://github.com/giantswarm/dashboards/assets/12008875/b179184d-2bc2-4606-860c-e876be1de165)
![image](https://github.com/giantswarm/dashboards/assets/12008875/71795dbe-013c-44ab-824e-638d6830d200)


<!--
Changelog must always be updated.
-->

### Checklist

- [x] Update changelog in CHANGELOG.md in an end-user friendly language.

QuentinBisson commented:
Looks good, would you maybe ask in area-kaas or sig-dev for inputs ?

fiunchinho commented:
The `kubebuilder` project already provides dashboards for controllers using `controller-runtime`, [have you seen this](https://master.book.kubebuilder.io/plugins/grafana-v1-alpha.html)? Maybe we can use it instead, so we don't have to maintain our own dashboards.

hervenicol commented:
> The `kubebuilder` project already provides dashboards for controllers using `controller-runtime`, [have you seen this](https://master.book.kubebuilder.io/plugins/grafana-v1-alpha.html)? Maybe we can use it instead, so we don't have to maintain our own dashboards.

Oh, I hadn't seen that when I had looked for existing dashboards!
I don't think it would free us from maintaining our own dashboards, but could probably generate a nice dashboard base. Or at least ideas on how to best use this data.

fiunchinho commented:
So now what are we adding? only the dashboards generated by `kubebuilder init`? If that's the case, I'm fine with it.

hervenicol commented:
> So now what are we adding? only the dashboards generated by `kubebuilder init`? If that's the case, I'm fine with it.

This dashboard was manually created.
But I'll try to improve it with the suggested metrics from the `kubebuilder init` dashboard.

hervenicol commented:
I added these panels, based on kube-builder dashboard generator's doc.
![image](https://github.com/giantswarm/dashboards/assets/12008875/eb46e427-52e9-45e8-9601-418f1988dee4)
![image](https://github.com/giantswarm/dashboards/assets/12008875/336bf402-4a3e-4030-83f2-41400e335fab)
![image](https://github.com/giantswarm/dashboards/assets/12008875/79591f1b-ef69-4dba-a147-7e13f8522c0b)

...and updated the doc accordingly:
![image](https://github.com/giantswarm/dashboards/assets/12008875/61be2ed2-f67a-4fb4-b103-8b256366ad2c)
